content_delegate.js

```diff
@@ -25,24 +25,59 @@ ContentDelegate.prototype.findPacerDocIds = function() {
     return;
   }
 
+  // Not all pages have a case ID, and potentially there are
+  // corner-cases where there are links to documents on another case?
+  let page_pacer_case_id = this.pacer_case_id ||
+    this.recap.getPacerCaseIdFromPacerDocId(this.pacer_doc_id, function(){});
+
+  let docsToCases = {};
+
+  // xxx why isn't it this.url? (undefined)
+  let pacer_doc_id = PACER.getDocumentIdFromUrl(url);
+  if (pacer_doc_id && page_pacer_case_id) {
+    // console.info('RECAP debug: Z doc ' + pacer_doc_id + ' to ' + page_pacer_case_id);
+    docsToCases[pacer_doc_id] = page_pacer_case_id;
+  }
+
   for (let i = 0; i < this.links.length; i++) {
     let link = this.links[i];
     if (PACER.isDocumentUrl(link.href)) {
       let pacer_doc_id = PACER.getDocumentIdFromUrl(link.href);
       $(link).data('pacer_doc_id', pacer_doc_id);
       this.pacer_doc_ids.push(pacer_doc_id);
+
+      let onclick = link.getAttribute('onclick');
+      // CMECF provides extra information on Document Links (DLS?) in the
+      // goDLS() function of an onclick handler, e.g.:
+      // <a
+      //   href="https://ecf.mad.uscourts.gov/doc1/09518360046"
+      //   onclick="goDLS('/doc1/09518360046','153992','264','','','1','','');
+      //     return(false);">
+      //  95
+      // </a>
+      // The parameters are defined in the unminified js
+      //   https://ecf.flnd.uscourts.gov/lib/dls_url.js
+      // as:
+      //   function goDLS(hyperlink, de_caseid, de_seqno, got_receipt,
+      //     pdf_header, pdf_toggle_possible, magic_num, hdr) {
+
+      let goDLS = onclick.match(/^goDLS\('([^']*)','([^']*)','([^']*)','([^']*)','([^']*)','([^']*)','([^']*)','([^']*)'\)/);
+
+      if (goDLS[2]) {
```

Comment:
I think a comment explaining this bit here would be useful.

Reply:
Clarified via ["array destructuring"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Unpacking_values_from_a_regular_expression_match)